### PR TITLE
Rubin na

### LINKED ISSUE #226: Allow Rubin's rules to work in case of `se = NA`?
Sometimes one might want to get point estimates without SE estimation. 

For example, for a responder analysis the analysis model might be a logistic regression for a dichotomized endpoint. One might additionally want the estimated proportion of responders in the two groups and not just the adjusted difference between them in log-odds scale, which is a regression coefficient in the logistic regression used to study significance but not easily interpretable.

This could be done by simply allowing the (user-defined) analysis model to return `se = NA` for the proportions parameters. The pooling function should simply in this case return `NA` for SE, CIs and p-value but still return the pooled point estimate.

### PR DESCRIPTION
**Pull request by nociale**
Closes #226 .

### CODE REVIEW

**Comment by nociale** [comment]
@gowerc Sorry Craig I don't get what is the cause of the failure of the check..

**Comment by gowerc** [comment]
I think its a issue with one of the current PROD cran packages. I need to investigate it a bit more, either way its just a problem with the master branch if you just make the PR to develop you shouldn't see it hopefully

File: tests/testthat/test-analyse.R [hunk 1]
```diff
@@ -48,6 +48,26 @@ test_that("basic constructions of `analysis` work as expected",{
     )
     expect_true(validate(x))
 
+
+    x <- as_analysis(
+        results = list(
+            list(p1 = list("est" = 1, "df" = 4, "se" = NA)),
+            list(p1 = list("est" = 2, "df" = 3, "se" = NA))
+        ),
+        method = method_bayes(n_samples = 2)
```
Comment (gowerc) [inline tests/testthat/test-analyse.R:57]: Should this not error due to the different DF  ?

File: R/analyse.R [hunk 2]
```diff
@@ -459,11 +459,17 @@ validate_analyse_pars <- function(results, pars) {
...
             assert_that(
                 all(!is.na(vapply(results_unnested, function(x) x[[par]], numeric(1)))),
                 msg = sprintf("Parameter `%s` contains missing values", par)
             )
+        } else if(par == "se") {
+            assert_that(
+                all(!is.na(vapply(results_unnested, function(x) x[[par]], numeric(1)))) ||
+                all(is.na(vapply(results_unnested, function(x) x[[par]], numeric(1)))) ,
+                msg = sprintf("Parameter `%s` contains both missing and observed values", par)
+            )
```
Comment (gowerc) [inline R/analyse.R:472]: Should the df also be handled by the second else if condition ? As in we allow DF to be NA but surely we don't allow for a mix of  missing and non-missing ?

File: tests/testthat/test-pool.R [hunk 3]
```diff
@@ -120,9 +136,61 @@ test_that("pool", {
...
+    results_bayes <- as_analysis(
+        method = method_bayes(n_samples = 5000),
+        results =
+            lapply(
+                seq_len(5000),
+                function(x) runanalysis(sample(vals, size = n, replace = TRUE))
+            )
+    )
+    bayes <- pool(results_bayes)
+
```
Comment (gowerc) [inline tests/testthat/test-pool.R:184]: Might be worth running this with a few different options like alternative and conf.int just to make sure theres no interactions

File: R/pool.R [hunk 4]
```diff
@@ -107,7 +107,7 @@ pool <- function(
 #' either `"rubin"`, `"jackknife"` or "`bootstrap"`.
 get_pool_components <- function(x) {
     switch(x,
-           "rubin" = c("est", "df", "se"),
+           "rubin" = c("est", "se", "df"),
```
Comment (gowerc) [inline R/pool.R:110]: Any reason for the switch here ?

File: tests/testthat/test-analyse.R (hunk 1 above)
Comment (nociale) [reply tests/testthat/test-analyse.R:57]: I think we are not verifying that all the df are the same in `validate.rubin` of the analysis output.

File: R/analyse.R (hunk 2 above)
Comment (nociale) [reply R/analyse.R:472]: Agreed!

File: R/pool.R (hunk 4 above)
Comment (nociale) [reply R/pool.R:110]: No :( , I think I changed the order by mistake

File: tests/testthat/test-pool.R (hunk 3 above)
Comment (nociale) [reply tests/testthat/test-pool.R]: Ok!

**Review by gowerc — APPROVED** [review]

**Pull request merged**